model/project.go

```diff
@@ -345,14 +345,15 @@ type TaskGroup struct {
 	Name string `yaml:"name" bson:"name"`
 
 	// data about the task group
-	MaxHosts      int             `yaml:"max_hosts" bson:"max_hosts"`
-	SetupGroup    *YAMLCommandSet `yaml:"setup_group" bson:"setup_group"`
-	TeardownGroup *YAMLCommandSet `yaml:"teardown_group" bson:"teardown_group"`
-	SetupTask     *YAMLCommandSet `yaml:"setup_task" bson:"setup_task"`
-	TeardownTask  *YAMLCommandSet `yaml:"teardown_task" bson:"teardown_task"`
-	Timeout       *YAMLCommandSet `yaml:"timeout,omitempty" bson:"timeout"`
-	Tasks         []string        `yaml:"tasks" bson:"tasks"`
-	Tags          []string        `yaml:"tags,omitempty" bson:"tags"`
+	MaxHosts           int             `yaml:"max_hosts" bson:"max_hosts"`
+	SetupGroupFailTask bool        	   `yaml:"setup_group_fail_task", bson:"setup_group_fail_task"`
```

Comment:
Tiniest nit and I'm open to other suggestions: Should the yaml be something like `setup_group_can_fail_task`. I'm trying to decide what the most obvious name for this would be to someone who has never seen it before.